lib/models/s3-event.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+const crypto = require('crypto');
+
+const S3Event = function (eventData, reqParams) {
+  const {reqHeaders,sourceIp} = reqParams;
+  let eventName = '';
+  const s3Object = {
+    "key": eventData.S3Item.key,
+    "sequencer": ('00' + (Math.floor(Date.now())).toString(16)).toUpperCase()
+  };
+  switch (eventData.eventType) {
+    case 'Copy':
+      eventName = 'ObjectCreated:Copy';
+      s3Object.size = eventData.S3Item.size;
+      break;
+
+    case 'Put':
+      eventName = 'ObjectCreated:Put';
+      s3Object.size = eventData.S3Item.size;
+      s3Object.eTag = eventData.S3Item.md5;
+      break;
+
+    case 'Post':
+      eventName = 'ObjectCreated:Post';
+      s3Object.size = eventData.S3Item.size;
+      s3Object.eTag = eventData.S3Item.md5;
+      break;
+
+    case 'Delete':
+      eventName = 'ObjectRemoved:Delete';
+      break;
+  }
+
+  const event = {
+    "Records": [
+      {
+        "eventVersion": "2.0",
+        "eventSource": "aws:s3",
+        "awsRegion": reqHeaders.host,
+        "eventTime": eventData.S3Item.creationDate || new Date().toISOString(),
+        "eventName": eventName,
+        "userIdentity": {
+          "principalId": 'AWS:' + randomString(21).toUpperCase(),
+        },
+        "requestParameters": {
+          "sourceIPAddress": sourceIp
+        },
+        "responseElements": {
+          "x-amz-request-id": randomString(16).toUpperCase(),
+          "x-amz-id-2": crypto.createHash('sha256').update(reqHeaders.host).digest('base64')
+        },
+        "s3": {
+          "s3SchemaVersion": "1.0",
+          "configurationId": randomString(8) + '-' + randomString(4) + '-' + randomString(4) + '-' + randomString(4) + '-' + randomString(12),
+          "bucket": {
+            "name": eventData.bucket,
+            "ownerIdentity": {
+              "principalId": randomString(14).toUpperCase()
+            },
+            "arn": "arn:aws:s3: : :" + eventData.bucket
+          },
+          "object": s3Object
+        }
+      }
+    ]
+  }
+  return event;
+};
+
+const randomString = function (length) {
```

Comment:
Apologies, I should have pointed this out in my last review. Instead of this function, can we use `crypto` for simplicity? Here's an example: https://stackoverflow.com/a/40191779/4743864